docs/API.md

```diff
@@ -0,0 +1,98 @@
+Pitaya API
+==========
+
+## Handlers
+
+Handlers are one of the core features of Pitaya, they are the entities responsible for receiving the requests from the clients and handling them, returning the response if the method is a request handler, or nothing, if the method is a push handler.
+
+### Signature
+
+Handlers must be public methods of the struct and have a signature following:
+
+Arguments
+* `context.Context`: the context of the request, which contains the client's session.
+* `pointer or []byte`: the payload of the request (_optional_).
+
+Push handlers return nothing, while request handlers must return:
+* `pointer or []byte`: the response payload
+* `error`: an error variable
+
+
+### Registering handlers
+
+Handlers must be explicitly registered by the application by calling `pitaya.Register` with a instance of the handler component. The handler's name can be defined by calling `pitaya/component`.WithName(`"handlerName"`) and the methods can be renamed by using `pitaya/component`.WithNameFunc(`func(string) string`).
+
+The clients can call the handler by calling `serverType.handlerName.methodName`.
+
+
+### Routing messages
+
+Messages are forwarded by pitaya to the appropriate server type, and custom routers can be added to the application by calling `pitaya.AddRoute`, it expects two arguments:
+
+* `serverType`: the server type of the target requests to be routed
+* `routingFunction`: the routing function with the signature `func(*session.Session, *route.Route, []byte, map[string]*cluster.Server) (*cluster.Server, error)`, it receives the user's session, the route being requested, the message and the map of valid servers of the given type.
```

Comment:
Please clarify how should be the map of valid servers. Is the key a server id or type?